src/snowflake/snowpark/_internal/data_source/utils.py

```diff
@@ -326,6 +327,8 @@ def worker_process(
                 stop_event,
             )
         except queue.Empty:
+            # indicate whether a process is exit gracefully
+            process_or_thread_error_indicator.put((os.getpid(),))
```

Comment:
wouldn't getpid return the same value in the multi thread approach?

Reply:
i modified the script to reuse existing queue, I also apply this logic only when using multi process